apps/cpp_rpc/CMakeLists.txt

```diff
@@ -32,6 +32,14 @@ if (OS)
    endif()
 endif()
 
+if(USE_OPENCL)
```

Comment:
tried to build this using follow command line `cmake  -DCMAKE_TOOLCHAIN_FILE=Android/Sdk/ndk/22.1.7171670/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_NATIVE_API_LEVEL=android-23 -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON -DANDROID_STL=c++_static ..` and got follow error:
```
CMake Error at apps/cpp_rpc/CMakeLists.txt:37 (if):
  if given arguments:

    "VERSION_LESS" "23"

  Unknown arguments specified
```
cmake version: 3.18.4 on ubuntu 21.10

Reply:
Does CMake 3.18.4 have a bug? [`VERSION_LESS`](https://cmake.org/cmake/help/latest/command/if.html#version-comparisons) does not seem to be a new CMake feature.

Reply:
I agree that it must not be new, on the other hand I pointed above what I observe. Have you tried to reproduce issue on your side? Not through docker that we do not have access

Reply:
I have no issue with CMake 3.20.6 + NDK 21.4.7075529 using

```cmake
if(${ANDROID_NDK_MAJOR} VERSION_LESS "23")
```

Note that there is no `$ENV`. Could you give it try?

Nor did I have to specify `ANDROID_NDK_MAJOR` with this in the cmake config.